src/speechless/benchmark.py

```diff
@@ -0,0 +1,166 @@
+from enum import Enum
+import re
+from rouge import Rouge
+from logging import Logger
+from argparse import ArgumentParser
+
+from speechless.processing.tokenization import EditToken
+from speechless.processing.analysis.tfidf import TfidfAnalysis
+from speechless.utils.logging import NULL_LOGGER
+from speechless.utils.cli import cli_subcommand
+
+
+class TokenGranularity(Enum):
+    WORD = 0
+    SENTENCE = 1
+
+# dataset format: %[bad sentence%], %(bad_word%) 
+class Benchmark:
+
+    def __init__(self, dataset_file : str, method : str, granularity : TokenGranularity, logger: Logger = NULL_LOGGER):
+        self.method = method
+        self.granularity = granularity
+        with open(dataset_file) as f:
+            self.data = f.read().replace('\n', '')
+            self.data = re.sub(' +', ' ', self.data)
+
+    def run(self):
+        expected_labels, gold_text = self._parse_dataset()
+        self._process_data_to_classify()
+
+        tokens = self._tokenize(self.data)
+
+        # TODO: extend for more methods
+        tfidf = TfidfAnalysis("text8", 0.1)
+        float_labels = tfidf.score_transcription(tokens)
+        actual_labels = []
+        for label in float_labels:
+            if label > 0.5:
+                actual_labels.append(0)
+            else:
+                actual_labels.append(1)
+
+        actual_text = self._get_text_from_labels(tokens, actual_labels)
+
+        print(actual_text)
+
+        rouge_res, labels_res = self._get_results(expected_labels, actual_labels, gold_text, actual_text)
+        print("Rouge result: ")
+        print(rouge_res)
+        print("Comparing labels result: ")
+        print(labels_res)
+
+    def _get_text_from_labels(self, tokens, labels):
+        text = ""
+        assert len(tokens) == len(labels)
+        for i in range(len(tokens)):
+            if labels[i] == 1:
+                text += " " + tokens[i].text
+        return text
+        
+    def _label_text(self, erase_regex, bad_text_regex):
+        text_to_label = re.sub(erase_regex, '', self.data)
+        text_to_label = re.sub(' +', ' ', text_to_label)
+        ranges = re.split(bad_text_regex, text_to_label)
+
+        labels = []
+        good_part = True
+        for i in ranges:
+            [labels.append(int(good_part)) for x in range(len(self._tokenize(i)))]
```

Comment:
What is `i` and `ranges`. Could you rename these, to make it more clear what is happening in this loop?